src/controllers/things_controller.js

```diff
@@ -395,74 +400,126 @@ ThingsController.ws('/:thingId/', (websocket, request) => {
   const thingId = request.params.thingId;
   const subscribedEventNames = {};
 
-  function sendMessage(message) {
-    websocket.send(message, (err) => {
+  async function sendMessage(message) {
+    // Convert from internal id to external id
```

Comment:
I don't understand the "internal" vs "external" ID bit. Can you explain?

Reply:
It's described in more detail in later comments but the gist is that all the models/internal things use `virtual-thing-1` as the id and the external thing description uses `https://cool-gateway.mozilla-iot.org/things/virtual-thing-1`